plugins/europeana/entity.js

```diff
@@ -50,4 +67,102 @@ function getEntityId(url) {
 }
 
+/**
+ * Retrieves the path for the entity, based on id and title
+ * @param {string} id of entity
+ * @param {string} title of entity
+ * @return {string} path
+ */
+export function getEntityPath(id, title) {
+  const entityId = id.split('-')[0];
+  const path = entityId + (title ? '-' + title.toLowerCase().replace(/ /g, '-') : '');
+  return path;
+}
+
+/**
+ * Search for specific facets for this entity to find the related entities
+ * @param {string} type of entity
+ * @param {string} id of entity
+ * @param {Object} params additional parameters sent to the API
+ * @return {Object} related entities
+ */
+export function relatedEntities(type, id, params) {
+  return axios.get('https://api.europeana.eu/api/v2/search.json', {
+    params: {
+      wskey: params.wskey,
+      profile: 'facets',
+      facet: 'edm_agent,skos_concept',
+      query: `"http://data.europeana.eu/${getEntityTypeApi(type)}/base/${getEntityId(id)}"`,
+      rows: 0
+    }
+  })
+    .then((response) => {
+      return response.data.facets ? getEntityFacets(response.data.facets, getEntityId(id)) : [];
+    })
+    .catch((error) => {
+      const message = error.response ? error.response.data.error : error.message;
+      throw new Error(message);
+    });
+}
+
+/**
+ * Return the facets that include data.europeana.eu
+ * @param {Object} the facets retrieved from the search
+ * @param {String} id of the current entity
+ * @return {Object} related entities
+ * TODO: limit results
+ */
+function getEntityFacets(facets, currentId) {
+  let entities = [];
+  for (let facet of facets) {
+    for (let field of facet['fields']) {
+      if (field['label'].includes('http://data.europeana.eu') && field['label'].split('/').pop() !== currentId) {
+        entities.push(field['label']);
+      }
+    }
+  }
+  return getDataForEntities(entities);
+}
+
+/**
+ * Lookup data for the given list of entity URIs
+ * @param {Object} the entities retrieved from the facet search
+ * @return {Object} looked up entities data
+ */
+function getDataForEntities(entities) {
+  if (entities.length === 0) return;
+
+  const q = entities.join('"+OR+"');
+  return axios.get(`https://www.europeana.eu/api/entities/search?query=entity_uri:("${q}")`, {
+    params: {
+      wskey: entityAPIKey
+    }
+  })
+    .then((response) => {
+      return getRelatedEntityTitleLink(response.data.items.slice(0,10));
+    })
+    .catch((error) => {
+      const message = error.response ? error.response.data.error : error.message;
+      throw new Error(message);
+    });
+}
+
+/**
+ * Format the the entity data
+ * @param {Object} the lookuped data for entities
+ * @return {Object} entity links and titles
+ */
+function getRelatedEntityTitleLink(entities) {
+  let entityDetails = [];
+
+  for (let entity of entities) {
+    if (entity.prefLabel.en) {
+      const entityLink = '/entity/' + getEntityTypeHumanReadable(entity.type) + '/' + getEntityPath(entity.id.toString().split('/').pop(), entity.prefLabel.en);
```

Comment:
Instead of constructing the full link here (and in the page view), return the entity type and path as distinct properties, then in views link/redirect to the [named route](https://router.vuejs.org/guide/essentials/named-routes.html) "entity-type-all".